### PR TITLE
Remove check-constants plugin

### LINKED ISSUE #5967: Remove the `check-es2015-constants` plugin and move code
Continuing from the fix for https://github.com/babel/babel/issues/5728 done in https://github.com/babel/babel/pull/5930, we should move that logic out into the two places it actually needs to be used.

* ~~`transform-es2015-modules-*` - `import`ed variables are constant, so assignments to them should throw~~ (This is now handled in `babel-helper-modules`)
* `transform-block-scoping` - `const` variables are constants, so assignments to them should throw

We currently have a plugin that doesn't really correspond to one specific part of the spec, meaning someone looking in `block-scoping` would probably expect it the handling to be in there, but it is not currently.

### PR DESCRIPTION
**Pull request by maurobringolf**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #5967  
| Patch: Bug Fix?          | no
| Major: Breaking Change?  | no
| Minor: New Feature?      | no
| Tests Added + Pass?      | Yes
| Documentation PR         | 
| Any Dependency Changes?  |
| License                  | MIT

This is a recreation of #6661 including the recent changes to `check-constants`.

As discussed in the issue, I moved the code from `check-constants` into `block-scoping` including the tests. I also removed all references to the plugin I could find.

I had to pass `state` into the `BlockScoping` class to access `addHelper`. I am not sure if this is the best way to do it, but it seems to work so far.

### CODE REVIEW

**Comment by maurobringolf** [comment]
nicolo-ribaudo Sorry for the delay. I'd still like to remove the check-constants plugin so I rebased the PR. The CI passed (98fa9ab) but after I moved the constants check into a separate method it failed due to timeout. Locally it passes.

Any ideas? Does this way trigger redundant constant check runs through scopes and make things slower than before? I thought the `DONE` set prevents that.

**Comment by existentialism** [comment]
@maurobringolf unlikely related to your PR/changes, restarted the build on circle

**Comment by maurobringolf** [comment]
Ahh nice, thank you! I didnt ask because the last time I did, it actually was an error in my code 😁 .

File: packages/babel-plugin-check-constants/README.md [hunk 1]
```diff
@@ -1,58 +0,0 @@
-# @babel/plugin-check-constants
-
-> Validate ES2015 constants (prevents reassignment of const variables).
```
Comment (hzoo) [inline packages/babel-plugin-check-constants/README.md]: next pr: we should maybe include parts of this readme into the blocking scoping one?

Comment (maurobringolf) [reply packages/babel-plugin-check-constants/README.md]: Good point, I can do this later tonight (today for you 😁)

Comment (maurobringolf) [reply packages/babel-plugin-check-constants/README.md]: @hzoo I just added a note for the constants check in the readme #7279. Improvements to wording welcome! Anything else that is needed from a documentation perspective?

**Pull request merged**